compiler/src/dotty/tools/dotc/transform/TypeTestsCasts.scala

```diff
@@ -105,10 +105,9 @@ object TypeTestsCasts {
       debug.println("P1 : " + P1.show)
       debug.println("X : " + X.show)
 
-      P1 <:< X       // constraint P1
+      P1 <:< X
 
-      // use fromScala2x to avoid generating pattern bound symbols
-      maximizeType(P1, span, fromScala2x = true)
+      maximizeType(P1, span, fromScala2x = false)
```

Comment:
Similarly, why is it correct to use `maximizeType` here ?

Reply:
Maximization of the type means we try to cover all possible values which conform to the skeleton `pre.F[_]` and `X`. Then we have to make sure all of them are actually of the type `P`,  which implies that the type arguments in `P` are trivial (no runtime check needed).